biz.aQute.launcher/src/aQute/launcher/Launcher.java

```diff
@@ -1199,16 +1205,39 @@ public void report(PrintStream out) {
 					out.println("Id    Levl State Modified      Location");
 
 					for (int i = 0; i < bundles.length; i++) {
+						String lastModified = "<>";
 						String loc = bundles[i].getLocation();
-						loc = loc.replaceAll("\\w+:", "");
-						File f = new File(loc);
+						try {
+							URI locURI = new URI(loc);
+							int count = 0;
+							WHILE: while (locURI.getScheme() != null) {
+								switch (locURI.getScheme()
+									.toLowerCase()) {
+									case "file" :
+										Path p = Paths.get(locURI);
+										long time = Files.getLastModifiedTime(p)
+											.toMillis();
+										lastModified = FORMAT.format(Files.getLastModifiedTime(p)
+											.toInstant());
+										loc = p.toString();
+										break WHILE;
+									case "reference" :
```

Comment:
I think this can be the default case? This will allow you to handle other schemes that do the same trick like I think `jar:`?

Reply:
Ok.

Reply:
I gave this a solid try, but I don't think it's feasible. When I make it the default, I have to special-case all of the schemes that we know don't refer to files on the local filesystem. Otherwise (eg) it resolves http://server/some/path to the local filesystem UNC path //server/some/path, and we end up with the same bug that triggered all of this.

The only easy way around this is to special-case all of the exceptions that ought not to be recursively resolved, and don't try and recursively resolve them. But if I have to list a bunch of special cases anyway, I think it's safer to special-case all the schemes that do the same trick rather than try and special-case all those that don't (opt-in rather than opt-out).